cedar-policy-validator/src/types/capability.rs

```diff
@@ -57,18 +58,52 @@ impl<'a> CapabilitySet<'a> {
 #[derive(Hash, Eq, PartialEq, Debug, Clone)]
 pub struct Capability<'a> {
     /// For this expression
-    on_expr: ExprShapeOnly<'a>,
+    on_expr: ExprShapeOnly<'a, ()>,
     /// This attribute is known to exist on that expression
-    attribute: &'a str,
+    ///
+    /// This expression represents the attribute name. It should have type string.
+    /// Often, but not necessarily, this is a string constant.
```

Comment:
Maybe something like: "Often this is a string constant, but in the case of tags it can be an expression"?

Reply:
Sounds great, updated